operator/backrest/backup.go

```diff
@@ -284,3 +264,60 @@ func CleanBackupResources(restclient *rest.RESTClient, clientset *kubernetes.Cli
 		}
 	}
 }
+
+// getCommandOptsFromPod adds command line options from the primary pod to a backrest job.
+// If not already specified in the command options provided in the pgtask, add the IP of the
+// primary pod as the value for the "--db-host" parameter.  This will ensure direct
+// communication between the repo pod and the primary via the primary's IP, instead of going
+// through the primary pod's service (which could be unreliable). also if not already specified
+// in the command options provided in the pgtask, then lookup the primary pod for the cluster
+// and add the PGDATA dir of the pod as the value for the "--db-path" parameter
+func getCommandOptsFromPod(clientset *kubernetes.Clientset, task *crv1.Pgtask,
+	namespace string) (commandOpts string, err error) {
+
+	// lookup the primary pod in order to determine the IP of the primary and the PGDATA directory for
+	// the current primaty
+	selector := fmt.Sprintf("%s=%s,%s in (%s,%s)", config.LABEL_PG_CLUSTER,
+		task.Spec.Parameters[config.LABEL_PG_CLUSTER], config.LABEL_PGHA_ROLE,
+		"promoted", "master")
+	pods, err := kubeapi.GetPods(clientset, selector, namespace)
+	if err != nil {
+		return
+	} else if len(pods.Items) > 1 {
+		err = fmt.Errorf("More than one primary found when creating backrest job %s",
+			task.Spec.Parameters[config.LABEL_JOB_NAME])
+		return
+	} else if len(pods.Items) == 0 {
+		err = fmt.Errorf("Unable to find primary when creating backrest job %s",
+			task.Spec.Parameters[config.LABEL_JOB_NAME])
+		return
+	}
+	pod := pods.Items[0]
+
+	var cmdOpts []string
+	var backrestPgHostRegex = regexp.MustCompile("--db-host|--pg1-host")
+	var backrestPgPathRegex = regexp.MustCompile("--db-path|--pg1-path")
```

Comment:
Consider defining these at the top of the file, given we just need them to compile once.